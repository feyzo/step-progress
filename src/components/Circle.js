import React from 'react';

function Circle({ stroke, strokeWidth = 5, radius = 10 }) {
  const center = radius + (strokeWidth / 2);
  const width = (radius * 2) + strokeWidth;

  return (
    <svg width={ width } height={ width }>
      <circle
        cx={center}
        cy={center}
        r={radius}
        stroke={stroke}
        strokeWidth={strokeWidth}
        fill="none"
      />
    </svg>
  );
}

export default Circle;
