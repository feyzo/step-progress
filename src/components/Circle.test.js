import React from 'react';
import renderer from 'react-test-renderer';

import Circle from './Circle';

test('Renders a circle', () => {
  const component = renderer.create(
    <Circle
      stroke={10}
    />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});