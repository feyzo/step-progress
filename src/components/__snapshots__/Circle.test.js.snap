exports[`test Renders a circle 1`] = `
<svg
  height={25}
  width={25}>
  <circle
    className="jYubib"
    cx={12.5}
    cy={12.5}
    r={10}
    stroke={10}
    strokeWidth={5} />
</svg>
`;
